### PR TITLE
Added Debug flag and fixed ios build issues

### PR DESCRIPTION
**Pull request by ztecharoberts**
This should allow people to ignore the debug pop up if desired. It also fixes ios issues with UUID.

### CODE REVIEW

File: src/android/IonicDeploy.java [hunk 1]
```diff
@@ -147,8 +148,10 @@ private String generateUDID() {
 
   private Boolean isDebug() {
     try {
-      if ((this.myContext.getPackageManager().getPackageInfo(this.myContext.getPackageName(), 0).applicationInfo.flags & ApplicationInfo.FLAG_DEBUGGABLE) != 0) {
-        return true;
+      if ((this.myContext.getPackageManager().getPackageInfo(this.myContext.getPackageName(), 0).applicationInfo.flags & ApplicationInfo.FLAG_DEBUGGABLE) != 0 ) {
+        if(this.shouldDebug != "false"){
```
Comment (timbru31) [inline src/android/IonicDeploy.java]: you could combine both if statements to one.

Comment (timbru31) [reply src/android/IonicDeploy.java]: Important: to compare a String. **Always** use .equals(). This code won't work.

File: src/ios/IonicDeploy.m [hunk 2]
```diff
@@ -51,13 +52,17 @@ + (BOOL) isPluginUpdating {
 }
 
 - (NSString *) generateUDID {
-    return [[NSUUID] UUIDString];
+    return [[NSUUID UUID] UUIDString];
 }
 
 - (BOOL) isDebug {
 #ifdef DEBUG
+if([self.shouldDebug isEqualToString:@"false"]){
```
Comment (timbru31) [inline src/ios/IonicDeploy.m]: add space after the parentheses

File: src/ios/IonicDeploy.m [hunk 3]
```diff
@@ -51,13 +52,17 @@ + (BOOL) isPluginUpdating {
...
+    return NO;
+} else{
```
Comment (timbru31) [inline src/ios/IonicDeploy.m]: add a space after else

File: plugin.xml [hunk 4]
```diff
@@ -17,6 +17,7 @@
 
     <preference name="APP_ID" />
     <preference name="CHANNEL_NAME" />
+    <preference name="DEBUG" />
```
Comment (timbru31) [inline plugin.xml]: I'd vote to add a default of "true"

File: README.md [hunk 5]
```diff
@@ -17,6 +17,7 @@ The plugin will be available on `window` as `IonicCordova`
 
 * `APP_ID` **Required** - Your Ionic Pro app ID
 * `CHANNEL_NAME` **Required** - The channel to check for updates from
+* `DEBUG` - Set false if you do not want the check when apk/ipa is in debug mode.
```
Comment (timbru31) [inline README.md]: I'd vote to to set the default to true.

File: src/android/IonicDeploy.java (hunk 1 above)
Comment (ztecharoberts) [reply src/android/IonicDeploy.java]: The idea here is that it doesnt have to be set. The only way it matters is if you set it to false.

**Comment by mlynch** [comment]
Some of these fixes landed in 561994a628c1d74b5c8bf6ea40716fd8f33e73d4. Will take a look at the Android side

**Comment by mlynch** [comment]
Okay so it looks like all the build issues were resolved in 561994a628c1d74b5c8bf6ea40716fd8f33e73d4, now this PR is for debugging, which I'll leave up to @Fuiste to process

**Comment by Fuiste** [comment]
The UUID changes should never have been committed, as they were for a feature that is no longer within the scope of this plugin, and no release was made which included them . 

The rest of this looks good to me though, and I'll release it in the next version of the plugin.

**Pull request merged**